src/Command/ModifyEntity/LineItem/LineItemChangeDatesCommand.php

```diff
@@ -0,0 +1,341 @@
+
+                $this->lineItemRepository->persist($lineItem);
+
+                $this->logger->info(
+                    sprintf(
+                        'New dates were set for line item with: "%d"',
+                        $lineItem->getId()
+                    ),
+                    $lineItem->jsonSerialize()
+                );
+            }
+
+            $messageTemplate = $this->isDryRun
+                ? '[DRY RUN] %d line item(s) have been updated.'
+                : '%d line item(s) have been updated.';
+
+            $this->symfonyStyle->success(sprintf($messageTemplate, $lineItemsCollection->count()));
+
+            if (!$this->isDryRun && $this->getApplication() !== null) {
+                $this->lineItemRepository->flush();
+            }
+
+            return 0;
+        } catch (Throwable $exception) {
+            $this->symfonyStyle->error(sprintf('An unexpected error occurred: %s', $exception->getMessage()));
+
+            return 1;
+        }
+    }
+
+    private function initializeLineItemIdsOption(InputInterface $input): void
+    {
+        $lineItemIds = array_filter(
+            explode(',', (string)$input->getOption(self::OPTION_LINE_ITEM_IDS)),
+            static function (string $value): bool {
+                return !empty($value) && (int)$value > 0;
+            }
+        );
+
+        if (empty($lineItemIds)) {
+            throw new InvalidArgumentException(
+                sprintf("Invalid '%s' option received.", self::OPTION_LINE_ITEM_IDS)
+            );
+        }
+
+        $this->lineItemIds = array_map('intval', $lineItemIds);
+    }
+
+    /**
+     * @throws InvalidArgumentException
+     */
+    private function initializeLineItemSlugsOption(InputInterface $input): void
+    {
+        $this->lineItemSlugs = array_filter(
+            explode(',', (string)$input->getOption(self::OPTION_LINE_ITEM_SLUGS)),
+            static function ($value): bool {
+                return !empty($value) && is_string($value) && mb_strlen($value) > 1;
+            }
+        );
+
+        if (empty($this->lineItemSlugs)) {
+            throw new InvalidArgumentException(
+                sprintf("Invalid '%s' option received.", self::OPTION_LINE_ITEM_SLUGS)
+            );
+        }
+    }
+
+    private function getFindLineItemCriteria(): FindLineItemCriteria
+    {
+        $criteria = new FindLineItemCriteria();
+
+        if (!empty($this->lineItemIds)) {
+            $criteria->addLineItemIds(...$this->lineItemIds);
+        }
+
+        if (!empty($this->lineItemSlugs)) {
+            $criteria->addLineItemSlugs(...$this->lineItemSlugs);
+        }
+
+        return $criteria;
+    }
+
+    /**
+     * @throws InvalidArgumentException
+     */
+    private function initializeDates(InputInterface $input): void
+    {
+        $inputStartDate = (string)$input->getOption(self::OPTION_START_DATE);
+        $inputEndDate = (string)$input->getOption(self::OPTION_END_DATE);
```

Comment:
Not sure that this two values are really needed there. Because for now their only purpose is to be passed to methods.
What I would propose is next:
a) leave those two variables in here;
b) combine two methods below into one, just returning the date (or, maybe, also throwing an error);
c) set dates into class properties in this method, because, now, this method is not initializing anything.
d) rename this one into smth liek `validateAndSetDates` (this naming is bad IMO, it's just a first thought), and method combined from two below into smth like `initializeDateConstraint`
e) *fully optional*, if I haven't missed such check above -> really validate dates, like `startDate` cannot be higher than `endDate`, etc. It shouldn't break anything, even without that type of check, but looks like a good way to shoot someones leg

Reply:
`$inputStartDate` and `$inputEndDate` are need just to display what was the real input from user when displaying the error message. Also, we have validation for end date before start date.

For the rest please see the explanation on https://github.com/oat-sa/simple-roster/pull/179#discussion_r545722906

I could move what is done on the `initializeDates` to the main `initialize` method, but it would also complain about cyclomatic complexity. There is not much escape here since we need to initialize/check then individually.

Reply:
Applied changes by @uncleempty, Thanks!